src/lake/lake.cpp

```diff
@@ -74,8 +82,12 @@ class LakeWriter {
     unordered_map<Oid, unordered_map<string, FileInfo>> xact_state;
 };
 
+// TODO(mooncake): Make `lake_writer` thread-safe. For the current implementation, concurrent table creation is not
```

Comment:
Correct me if wrong, I don't see any synchronization primitives used against `lake_writer` and its internal states, thus potential data race.

Reply:
I believe `LakeWriter` is not accessed by multiple threads concurrently right now
We haven't parallelized `ColumnstoreInsert`, `ColumnstoreDelete`, or `ColumnstoreUpdate` yet (`ParallelSink()` defaults to false)

Reply:
On the other hand, `CREATE TABLE` and `COMMIT` is single-threaded because Postgres is process-based, but not thread-based

Reply:
That makes sense, TODO removed.